### PR TITLE
iocage_legacy UCL config (develop branch) support

### PR DESCRIPTION
**Pull request by gronke**
- [x] can list properties of a UCL jail created with iocage_legacy@1.7.6
- [x] can set properties of a UCL jail created with iocage_legacy@1.7.6
- [x] can start nullfs basejails created with iocage_legacy@1.7.6
- [x] can start standalone jails created with iocage_legacy@1.7.6
- [x] can create new standalone jails using a release fetched with iocage_legacy#develop
- [x] can create new zfs basejail using a release fetched with iocage_legacy#develop
- [x] iocage_legacy@1.7.6 can create jails from releases fetched with libiocage
- [x] jail config rlimits=no disables rlimits
- [x] can read defaults from UCL file automatically
- ~can read defaults from UCL config file as `ioc get all defaults@ucl`~
- ~can write defaults to UCL config file as `ioc set key=value defaults@ucl`~

### CODE REVIEW

**Comment by gronke** [comment]
iocage_legacy master mounts a few additional datasets that do not get unmounted properly with libiocage.
```
/iocage/releases/11.1-RELEASE/root/etcupdate on /iocage/jails/7d5b9383-ccb7-11e7-95ce-589cfc0bc5b6/root/etcupdate (nullfs, local, read-only)
/iocage/releases/11.1-RELEASE/root/usr/ports on /iocage/jails/7d5b9383-ccb7-11e7-95ce-589cfc0bc5b6/root/usr/ports (nullfs, local, read-only)
/iocage/releases/11.1-RELEASE/root/usr/src on /iocage/jails/7d5b9383-ccb7-11e7-95ce-589cfc0bc5b6/root/usr/src (nullfs, local, read-only)
tmpfs on /iocage/jails/7d5b9383-ccb7-11e7-95ce-589cfc0bc5b6/root/tmp (tmpfs, local)
```

**Comment by gronke** [comment]
Asciinema-Demo: https://asciinema.org/a/RWZuADoTzOZTI9DJ1jBH8aJvh

File: iocage/lib/Jail.py [hunk 1]
```diff
@@ -1145,6 +1152,23 @@ def _teardown_mounts(self) -> None:
...
+        if self.config.legacy is True:
+            command = [
+                "/bin/sh",
+                "-c",
+                " | ".join([
+                    "mount",
+                    f"grep 'on {self.root_dataset.mountpoint}'",
+                    "grep nullfs",
+                    "cut -f1 -d ' '",
+                    "xargs -n1 umount"
```
[the remark this answers is not in the document]
Comment (igalic) [reply iocage/lib/Jail.py]:
```sh
mount -t nullfs | sed -r 's/(.+) on (.+) \(nullfs, .+\)$/\2/' | grep "^/iocage/jails/foo/" | xargs echo umount 
```
should get us there quicker. that is, with less pipes and execs.
however, the security hazard of parsing an arbitrary string format still remains and we should open an issue to start using a library that has direct access to the fstab structs.

**Comment by igalic** [comment]
so far so good, the only thing missing are snapshots…
perhaps we should split those out into a different pull request, since we don't have the capability of managing snapshots right now anyway?

File: iocage/lib/Host.py [hunk 2]
```diff
@@ -87,13 +89,25 @@ def __init__(
...
+        self._init_defaults(defaults)
+
+    def _init_defaults(
+        self,
+        defaults: typing.Optional[iocage.lib.Resource.DefaultResource]=None
+    ) -> None:
+
         if defaults is not None:
             self._defaults = defaults
+        else:
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/Host.py]: Not sure if I got your point. @igalic Can please you clarify?

File: iocage/lib/helpers.py [hunk 3]
```diff
@@ -266,6 +267,18 @@ def to_json(data: typing.Dict[str, typing.Any]) -> str:
     return str(json.dumps(output_data, sort_keys=True, indent=4))
 
 
+def to_ucl(data: typing.Dict[str, typing.Any]) -> str:
```
[the remark this answers is not in the document]
Comment (gronke) [reply iocage/lib/helpers.py]: It is similar to `to_json` which also exists as helper function. we could move both functions to their config package instead. (noted for later)

**Pull request merged**